recruitment/forms.py

```diff
@@ -37,103 +37,112 @@ class CompareForm(forms.Form):
-            application_list = [application for application in application_list if
-                                name.lower() in application.user.get_full_name().lower()]
-
-        programme = search_form.cleaned_data['programme']
-        if programme:
-            application_list = [application for application in application_list if
-                                programme == application.user.profile.programme]
-
-        registration_year = search_form.cleaned_data['registration_year']
-        if registration_year:
-            application_list = [application for application in application_list if
-                                int(registration_year) == application.user.profile.registration_year]
-
-        rating = search_form.cleaned_data['rating']
-        if rating:
-            application_list = [application for application in application_list if int(rating) == application.rating]
-
-        submission_date = search_form.cleaned_data['submission_date']
-        if submission_date:
-            application_list = [application for application in application_list if
-                                submission_date.lower() in date_filter(
-                                    application.submission_date + timezone.timedelta(hours=2), "d M H:i").lower()]
-
-        roles_string = search_form.cleaned_data['roles']
-        if roles_string:
-            application_list = [application for application in application_list if
-                                roles_string.lower() in application.roles_string().lower()]
-
-        interviewer = search_form.cleaned_data['interviewer']
-        if interviewer:
-            application_list = [application for application in application_list if
-                                interviewer == application.interviewer]
-
-        recommended_role = search_form.cleaned_data['recommended_role']
-        if recommended_role:
-            application_list = [application for application in application_list if
-                                recommended_role == application.recommended_role]
-
-        state = search_form.cleaned_data['state']
-        if state:
-            application_list = [application for application in application_list if state == application.state()]
-
-        return application_list
+	name = forms.CharField(widget=forms.TextInput(attrs={'class': 'form-control'}), required=False)
+
+	programme = forms.ModelChoiceField(
+		queryset=Programme.objects.all(),
+		widget=forms.Select(attrs={'class': 'form-control'}),
+		required=False
+	)
+
+	submission_date = forms.CharField(widget=forms.TextInput(attrs={'class': 'form-control'}), required=False)
+
+	role = forms.ModelChoiceField(
+		queryset=Role.objects.all(),
+		widget=forms.Select(attrs={'class': 'form-control'}),
+		required=False
+	)
+
+	priority = forms.ChoiceField(
+		# The choices will also be populated from the view, so this line might be superfluous
+		choices = [('', '-------'), (0, '1:st'), (1, '2:nd'), (2, '3:rd')],
```

Comment:
I have seen that in a previous commit this field was populated with 'Any', '1', '2', '3' matching the choices that are given in the view. It does not really matter but if it should be 'Any' etc. then it would be good to update it here as well to avoid confusement. :)

Reply:
I found the table to look more consistent with '------' so I opted for '-------', '1:st', '2:nd' etc but if anyone has another opinion on what looks best it would be super easy to change. :)

As the form is only ever used in this particular view, the choices are now populated only from there, see commit 06eb7189dee4f0206ab5cd10499c5e4552448bde.